### ISSUE TITLE
1.7.2 fails to build with stricter C compilers, consider new release

### ISSUE THREAD

**Issue by thesamesam** [libunwind/libunwind#672]
Modern C compilers are becoming stricter with a variety of changes over the last year or so.

GCC 14 in particular (to be released in ~April 2024) fails to build libunwind-1.7.2 like:
```
/var/tmp/portage/sys-libs/libunwind-1.7.2/work/libunwind-1.7.2/src/x86/Gos-linux.c: In function ‘_Ux86_local_resume’:
/var/tmp/portage/sys-libs/libunwind-1.7.2/work/libunwind-1.7.2/src/x86/Gos-linux.c:302:22: error: passing argument 1 of ‘_Ux86_sigreturn’ from incompatible pointer type [-Wincompatible-pointer-types]
  302 |       x86_sigreturn (sc);
      |                      ^~
      |                      |
      |                      struct sigcontext *
In file included from /var/tmp/portage/sys-libs/libunwind-1.7.2/work/libunwind-1.7.2/src/x86/Gos-linux.c:26:
/var/tmp/portage/sys-libs/libunwind-1.7.2/work/libunwind-1.7.2/src/x86/unwind_i.h:64:42: note: expected ‘unw_cursor_t *’ {aka ‘struct unw_cursor *’} but argument is of type ‘struct sigcontext *’
   64 | extern void x86_sigreturn (unw_cursor_t *cursor);
      |                            ~~~~~~~~~~~~~~^~~~~~
make[2]: *** [Makefile:3925: x86/Gos-linux.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
```

This can be emulated with `-Werror=incompatible-pointer-types -Werror=implicit -Werror=int-conversion` on an older GCC or Clang.

master builds fine for me at commit 75281907d37da9e85a1c610fda7584310ebe270f.

Could a new release be considered in the near future? Thank you!

**Comment by bertwesarg** [comment]
See #639

**Comment by bregma** [comment]
That particular failure was recently fixed by f35c38746d47a5159163ca9e8845a3ac8cf1bc53, which also fixed the last blocker for the 1.8 release.  The rest of the [tagged issues](https://github.com/libunwind/libunwind/milestone/2) can be rolled over to 1.9 so I'll branch and prep a new release candidate in the next few days as I find the time.

**Comment by bregma** [comment]
Marking this as completed since the build problems have been fixed in release 1.8.0.
